common4j/src/main/com/microsoft/identity/common/java/network/NetworkMarkerManager.java

```diff
@@ -0,0 +1,182 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.java.network;
+
+
+import com.microsoft.identity.common.java.logging.Logger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import lombok.NonNull;
+
+public class NetworkMarkerManager {
+    private static final String TAG = NetworkMarkerManager.class.getSimpleName();
+
+    private static final List<NetworkMarker> mNetworkMarkers = Collections.synchronizedList(new ArrayList<NetworkMarker>());
+    private static final Map<String, NetworkState[]> mNetworkMarkerOverride = Collections.synchronizedMap(new HashMap<String, NetworkState[]>());
+    private static final NetworkStatesHandler networkStatesHandler = new NetworkStatesHandler();
+
+    private static boolean mEnabled = false;
+    private static NetworkMarker mCurrentMarker = null;
+    private static INetworkStateChangeHandler mStateChangeHandler = null;
+    private static long mStartTime;
+
+    public NetworkMarkerManager() {
```

Comment:
Is there any reason why we need this empty public constructor?

Reply:
Not required. Thanks for noticing this. I'll remove it.